File: Models/CauHinhDht.cs
using System;
using System.Collections.Generic;

namespace WebAPI_NRW.Models.Database;

public partial class CauHinhDht
{
    public int Id { get; set; }

    public int? MaDoiTuong { get; set; }

    public int? MaDongHo { get; set; }

    public string? GhiChu { get; set; }

    public DateTime NgayTao { get; set; }

    public DateTime? NgayCapNhat { get; set; }

    public int NguoiTao { get; set; }

    public int? NguoiCapNhat { get; set; }

    public virtual DongHoTong? MaDongHoNavigation { get; set; }

    public virtual NguoiDung? NguoiCapNhatNavigation { get; set; }

    public virtual NguoiDung NguoiTaoNavigation { get; set; } = null!;
}

File: src/types/dong-ho-tong/cau-hinh-dht.ts
/* ========== REQUEST (CLIENT --> API) ========== */
// Thêm dữ liệu
export interface AddCauHinhDhtRequest {
  maDoiTuong?: number;
  maDongHo?: string; // giống BE: string
  ghiChu?: string;
  ngayTao: string;
  nguoiTao: string; // BE dùng string
}

// Cập nhật dữ liệu
export interface UpdateCauHinhDhtRequest {
  maDoiTuong?: number;
  maDongHo?: string; // giống BE
  ghiChu?: string;
  ngayCapNhat?: string;
  nguoiCapNhat?: string; // BE dùng string
}

/* ========== RESPONSE (API --> CLIENT) ========== */
// Dữ liệu trả về
export interface CauHinhDhtResponse {
  locked: unknown;
  id: number;
  maDoiTuong?: number;
  maDongHo?: string;
  ghiChu?: string;
  ngayTao: string;
  ngayCapNhat?: string;
  nguoiTao?: string;
  nguoiCapNhat?: string;
}
